### PR TITLE
Fix registration of applications with same names and ids but from different rank devices

### PR DESCRIPTION
**Pull request by dcherniev**
Added new method GetDeviceHandlerById into Service to get device handler by device identifier;
Fixed registration of applications with the same names and app ids but from different rank devices;
Removed incorrect check from ValidateDeviceInfo checking method.

### CODE REVIEW

**Comment by dcherniev** [comment]
Please review: @LuxoftAKutsan, @okozlovlux, @LevchenkoS, @dtrunov

File: src/components/application_manager/include/application_manager/core_service.h [hunk 1]
```diff
@@ -122,6 +122,13 @@ class CoreService : public Service {
   virtual void ResetAccess(const SeatLocation& zone, const std::string& module);
 
   /**
+   * Gets device handler for device with certain ID
+   * @param device_id the ID of the connected device
+   * @return device handler if device with requested ID was found
+   */
+  virtual uint32_t GetDeviceHandlerById(const std::string& device_id);
```
Comment (dtrunov) [inline src/components/application_manager/include/application_manager/core_service.h]: @dcherniev Please use OVERRIDE here.

File: src/components/can_cooperation/src/can_module.cc [hunk 2]
```diff
@@ -275,11 +275,12 @@ functional_modules::ProcessResult CANModule::HandleMessage(
               params.isMember(message_params::kRank) &&
               params[message_params::kRank].isString();
           if (valid) {
-            uint32_t device_id =
-                params[message_params::kDevice][json_keys::kId].asUInt();
+            std::string device_id =
+                params[message_params::kDevice][json_keys::kId].asString();
+            uint32_t device_handle = service()->GetDeviceHandlerById(device_id);
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/src/can_module.cc]: @dcherniev can be const

File: src/components/can_cooperation/src/can_module.cc [hunk 3]
```diff
@@ -300,16 +301,17 @@ functional_modules::ProcessResult CANModule::HandleMessage(
                   params.get(message_params::kDeviceLocation,
                              Json::Value(Json::nullValue)));
           if (valid) {
-            uint32_t device_id =
-                params[message_params::kDevice][json_keys::kId].asUInt();
+            std::string device_id =
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/src/can_module.cc]: @dcherniev can be const

File: src/components/can_cooperation/src/can_module.cc [hunk 4]
```diff
@@ -275,11 +275,12 @@ functional_modules::ProcessResult CANModule::HandleMessage(
               params.isMember(message_params::kRank) &&
               params[message_params::kRank].isString();
           if (valid) {
-            uint32_t device_id =
-                params[message_params::kDevice][json_keys::kId].asUInt();
+            std::string device_id =
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/src/can_module.cc]: @dcherniev can be const

File: src/components/application_manager/include/application_manager/core_service.h (hunk 1 above)
Comment (dcherniev) [inline src/components/application_manager/include/application_manager/core_service.h]: @dtrunov Fixed, please check  db859f5.

File: src/components/can_cooperation/src/can_module.cc (hunk 4 above)
Comment (dcherniev) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan Fixed, please check 9fa2677.

File: src/components/can_cooperation/src/can_module.cc (hunk 2 above)
Comment (dcherniev) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan Fixed, please check 9fa2677.

File: src/components/can_cooperation/src/can_module.cc (hunk 3 above)
Comment (dcherniev) [inline src/components/can_cooperation/src/can_module.cc]: @LuxoftAKutsan Fixed, please check 9fa2677.

**Pull request merged**